### PR TITLE
feat: Added Merge command

### PR DESCRIPTION
**Pull request by MrLalelu**
As I (and other #618) was looking for this feature for quite a while now, I just drafted it. I am not typescript, vscode & svn expert so am really happy for any improvement ideas.

### CODE REVIEW

File: src/commands/merge.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if (!branch) {
+      return;
+    }
+
+    await this.merge(repository, branch);
+  }
+
+  async merge(repository: Repository, branch: IBranchItem) {
+    let reintegrate = false;
+    if (repository.currentBranch == "trunk") {
```
Comment (JohnstonCode) [inline src/commands/merge.ts:24]: This will not work for people that don't call their main branch trunk

Comment (JohnstonCode) [reply src/commands/merge.ts:24]: https://github.com/JohnstonCode/svn-scm/blob/d7b26e69b5260d1dccbe6f1e241879a20b513e7e/src/helpers/branch.ts#L9

Comment (JohnstonCode) [reply src/commands/merge.ts:24]: This might help

Comment (MrLalelu) [reply src/commands/merge.ts]: Good point. Added a isTrunk function to branches, that uses the same logic as in getBranchName.

File: src/commands/merge.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  async merge(repository: Repository, branch: IBranchItem) {
+    let reintegrate = false;
+    if (isTrunk(repository.currentBranch)) {
+      reintegrate = true;
+    }
+
+    try {
+      await repository.merge(branch.path, reintegrate);
+    } catch (error) {
+      console.log(error);
```
Comment (JohnstonCode) [inline src/commands/merge.ts:31]: Remove this

**Review by JohnstonCode — CHANGES_REQUESTED** [review]

File: src/commands/merge.ts (hunk 2 above)
Comment (MrLalelu) [reply src/commands/merge.ts:31]: Sorry I don't get what you propose to remove. The entire try catch? or simply the log of the error? I guess the second, right? How about moving it to the last else block l.51, as there one has no clue what went wrong?

Comment (MrLalelu) [reply src/commands/merge.ts:31]: Then a log of the error could be useful.

Comment (JohnstonCode) [reply src/commands/merge.ts:31]: Yeah moving the console.log to the else block would be okay

**Review by JohnstonCode — APPROVED** [review]

**Comment by JohnstonCode** [comment]
fixes #618

**Comment by JohnstonCode** [comment]
@MrLalelu thanks for your work on this!

**Pull request merged**